main.cpp

```diff
@@ -318,15 +322,19 @@ FloatType& FloatType::operator*=(const float other)
 FloatType& FloatType::apply(std::function<FloatType&(float&)> floatFunc)
 {
     if(floatFunc)
-    floatFunc(*floatTypeHeap);
+    {
+        floatFunc(*floatTypeHeap);
```

Comment:
- [ ] return the result of calling the std::function if the std::function is valid.